### ISSUE TITLE
Which frame HIRONX.getCurrentPose refers to

### ISSUE THREAD

**Issue by 130s** [start-jsk/rtmros_hironx#212]
After https://github.com/start-jsk/rtmros_hironx/issues/45, we are now able to specify the frame for the pose of the eef to root from.

```
robot.setTargetPose(self, gname, pos, rpy, tm, ref_frame_name=None)
```

About get*Pose methods, `getCurrentPose` and `getReferencePose`, which frame do they refer to, is unclear.

This question rises from a user who owns both `Hiro` and `NEXTAGE OPEN` where the position of the link chain root is different in z direction (the model of NXO stands on a base structure when Hiro doesn't).

**Comment by 130s** [comment]
Confimed the phenomenon (and a workaround) in simulation.

On Hironx that does not come with a base,

```
term1$ rtmlaunch hironx_ros_bridge hironx_ros_bridge_simulation.launch
term2$ ipython -i `rospack find hironx_ros_bridge`/scripts/hironx.py
:
In [1]: robot.getCurrentPosition('LARM_JOINT5')
Out[1]: [0.32556275164378523, 0.18236394191140365, 0.07462472659364472]
```

Since NXO comes with base, position `z` is 1 meter taller.

```
term1$ rtmlaunch nextage_ros_bridge nextage_ros_bridge_simulation.launch 
term2$ ipython -i `rospack find nextage_ros_bridge`/script/nextage.py
:
In [3]: robot.Groups
Out[3]: 
[['torso', ['CHEST_JOINT0']],
 ['head', ['HEAD_JOINT0', 'HEAD_JOINT1']],
 ['rarm',
  ['RARM_JOINT0',
   'RARM_JOINT1',
   'RARM_JOINT2',
   'RARM_JOINT3',
   'RARM_JOINT4',
   'RARM_JOINT5']],
 ['larm',
  ['LARM_JOINT0',
   'LARM_JOINT1',
   'LARM_JOINT2',
   'LARM_JOINT3',
   'LARM_JOINT4',
   'LARM_JOINT5']]]

In [5]: robot.getCurrentPosition('LARM_JOINT5')
Out[5]: [0.3255751432930864, 0.1823632083754606, 1.0376247521028383]
```

Specifyin the reference frame like below would give the similar `z` value.

```
In [2]: robot.getCurrentPosition('LARM_JOINT5', 'WAIST')
Out[2]: [0.3255751432930864, 0.1823632083754606, 0.06762475210283836]
```

I'm not sure what's the best optimal solution. For now as a temporary fix, do we want to add this workaround to `NextageClient.getCurrentPosition` as follows (This way then we would lose the base height info)?
- Now: NextageClient.getCurrentPosition(lname=None, frame_name=None)
- New: NextageClient.getCurrentPosition(lname=None, frame_name='WAIST')

**Comment by k-okada** [comment]
> - Now: NextageClient.getCurrentPosition(lname=None, frame_name=None) 
> - New: NextageClient.getCurrentPosition(lname=None, frame_name='WAIST') 
> 
> This may good, but need to be release carefully (inform potential users)
> They may like to keep original configuration, but the height of WAIST (from 
> BASE) is not correct, because you can change the height of the physical 
> base with screw. So if they write program with current setup, they specify 
> the current position with respect to not the real floor but ad some offset, 
> so it's better to write motion with respect to WAIST (or need to change 
> model file to have exact BASE->WAIST length)

**Comment by 130s** [comment]
> They may like to keep original configuration, but the height of WAIST (from BASE) is not correct, because you can change the height of the physical base with screw.

True.

> so it's better to write motion with respect to WAIST (or need to change model file to have exact BASE->WAIST length)

I think we want to encourage developers to always refer to `WAIST` whenever possible.

That said, my suggestion still looks reasonable to me (with careful announcement to all possible users):

>   Now: NextageClient.getCurrentPosition(lname=None, frame_name=None)
>   New: NextageClient.getCurrentPosition(lname=None, frame_name='WAIST')

**Comment by 130s** [comment]
Addressed in #298.
